core/templates/pages/blog-admin-page/blog-admin-page.component.spec.ts

```diff
@@ -0,0 +1,427 @@
+
+        component.formData.updateRole.newRole = 'BLOG';
+        component.formData.updateRole.username = 'username';
+
+        expect(component.formData.updateRole.isValid()).toBe(false);
+
+        component.formData.updateRole.newRole = 'ADMIN';
+        component.formData.updateRole.newRole = '';
+
+        expect(component.formData.updateRole.isValid()).toBe(false);
+
+        component.formData.updateRole.newRole = null;
+        component.formData.updateRole.username = 'username';
+
+        expect(component.formData.updateRole.isValid()).toBe(false);
+
+        component.formData.updateRole.newRole = 'BLOG_ADMIN';
+        component.formData.updateRole.username = null;
+
+        expect(component.formData.updateRole.isValid()).toBe(false);
+      }));
+
+    it('should enable update role button if the input values are valid',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        component.formData.updateRole.newRole = 'BLOG_POST_EDITOR';
+        component.formData.updateRole.username = 'username';
+
+        expect(component.formData.updateRole.isValid()).toBe(true);
+      }));
+  });
+
+  describe('when removing a blog editor', () => {
+    it('should submit remove editor role form successfully', fakeAsync(() => {
+      component.ngOnInit();
+      tick();
+      component.formData.removeEditorRole.username = 'username';
+      spyOn(blogAdminBackendApiService, 'removeBlogEditorAsync')
+        .and.returnValue(Promise.resolve(Object));
+      spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(false);
+      component.submitRemoveEditorRoleForm(component.formData.updateRole);
+
+      expect(startTaskSpy).toHaveBeenCalled();
+      expect(component.statusMessage).toBe('Processing query...');
+
+      flushMicrotasks();
+
+      expect(component.statusMessage).toBe('Success.');
+      expect(finishTaskSpy).toHaveBeenCalled();
+    }));
+
+    it('should not submit remove editor role form if already' +
+      'a task is in queue', fakeAsync(() => {
+      component.ngOnInit();
+      tick();
+      component.formData.removeEditorRole.username = 'username';
+
+      spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(true);
+      component.submitRemoveEditorRoleForm(component.formData.updateRole);
+
+      expect(startTaskSpy).not.toHaveBeenCalled();
+      expect(finishTaskSpy).not.toHaveBeenCalled();
+    }));
+
+    it('should not submit remove editor role form in case of backend error',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        component.formData.removeEditorRole.username = 'username';
+        spyOn(blogAdminBackendApiService, 'removeBlogEditorAsync')
+          .and.returnValue(Promise.reject({
+            error: { error: 'Internal Server Error.'}
+          }));
+        spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(false);
+        component.submitRemoveEditorRoleForm(component.formData.updateRole);
```

Comment:
Am I right in assuming that this is the action step? if it is please add a new line to distinguish between the steps

Reply:
Done. yup right.